spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -188,6 +199,23 @@ data class SpdxCompoundExpression(
         right.validate(strictness)
     }
 
+    override fun validChoicesDnf(): Set<SpdxExpression> {
+        return when (operator) {
+            SpdxOperator.AND -> setOf(this)
+            SpdxOperator.OR -> {
+                val validChoicesLeft = when (left) {
```

Comment:
Why is the case distinction (ẁhen-clause) needed?

Reply:
It was actually a left-over from before I moved the helper function to `SdpxExpression`, but it turns out I cannot access `left.validChoicesForDnf()` here because it is protected, so it's still required.